packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -167,19 +212,30 @@ export const SelectAssetModal = ({
                     innerAddon={<Icon name="search" variant="tertiary" size="medium" />}
                     innerAddonAlign="left"
                 />
-                <NetworkTabs
-                    networkCategories={networkCategories}
-                    networkCount={networkCount}
-                    searchCategory={searchCategory}
-                    setSearchCategory={setSearchCategory}
-                />
+                {filterCategories?.categoriesType === 'networks' && (
+                    <NetworkTabs
+                        networkCategories={filterCategories.categories}
+                        networkCount={networkCount}
+                        searchCategory={searchCategory}
+                        setSearchCategory={setSearchCategory}
+                    />
+                )}
+                {filterCategories?.categoriesType === 'tokens' &&
+                    setPickedSendTokenCategory &&
+                    pickedSendTokenCategory && (
+                        <SendTokenTabs
+                            sendTokenCategories={filterCategories.categories}
+                            pickedCategory={pickedSendTokenCategory}
+                            setPickedCategory={setPickedSendTokenCategory}
+                        />
+                    )}
```

Comment:
Ad in person discussion with @tomasklim : this is what I was talking about - this should be rendered above (in coinmarket or send form respectively), and passed to `SelectAssetModal` as JSX.
Like `children`, but in this case I'd name the prop `categoryTabs: ReactNode` or something, because `children` would be misleading name (it's just specific part of the content, not whole content).
:arrow_right: it will make this component dumber, and that's how I like it.

I think it's possible here; both `NetworkTabs` and `SendTokenTabs` use only props from above, or values directly calculated from them. But if I'm mistaken, disregard.

Reply:
Oh no, the `searchCategory` comes from here. That's a mess :see_no_evil:

Reply:
I totally agree

Reply:
done